File: user_pgm.java
import java.io.*;
import java.util.Scanner;

public class user_pgm {


    /* This method should contain all user code. You get strings representing the
       state of the code (some number of chars; '_' if the letter in that space has
       not been guessed yet or if it has been guessed, the correct character shows
       in that place). Example: the word is "apple" and 'a' and 'l' were previous
       correct guesses, then the state would look like "a__l_". guesses is a string
       of all guesses in no particular order. The letters will not have any separator.
       numGuesses is an int representing the number of previous guesses. This value
       will most likely not be useful to you, but I included it just in case.
    */
    public static char userCode(String state, String guesses, int numGuesses) {
        
    }
    
    public static void printChar(char toSend) {
	System.out.println(Character.toLowerCase(toSend));
	System.exit(0);
    }
    
    public static void main(String[] args) {
	Inputs inputs = new Inputs();
	inputs.setInputs();
	
	printChar(userCode(getState(), getGuesses(), getNumGuesses()));
    }

    public static class Inputs {
	String state, guesses;
	int numGuesses;

	public void setInputs() {
	    Scanner input = new Scanner(System.in);

	    state = input.next();
	    guesses = input.next();
	    numGuesses = input.nextInt();

	    input.close();
	}

	public String getState() {
	    return state;
	}

	public String getGuesses() {
	    return guesses;
	}

	public int getNumGuesses() {
	    return numGuesses;
	}
	
    }
    
}

File: user_pgm.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

void get_in(char* state, char* guesses, int* x) {
  state = malloc(63);
  guesses = malloc(63);

  scanf("%s", state);
  scanf("%s", guesses);
  scanf("%d", x);
}

void print_char(char to_send) {
  printf("%c", tolower(to_send));

  exit(0);
}

/* 
 * This function should contain all of your code. It gets the strings state and guesses
 * and the int num_guesses as parameters. state is a string representing the current
 * state of the secret word. Any characters in the string that have not been correctly
 * guessed yet are represented as '_'. Example: if the word to be guessed is "apple",
 * and you have previously guessed 'a' and 'l', the string will be "a__l_" (ending with
 * a null terminating byte). guesses is a character array of the previously guessed
 * characters ending in a null terminating byte. The return value should be the
 * character that you want to guess.
 */
char user_func(char* state, char* guesses, int num_guesses) {

  /* YOUR CODE HERE */
  
}

int main(void) {
  char *state, *guesses, to_send;
  int num_guesses;

  get_in(state, guesses, &num_guesses);

  print_char(user_func(state, guesses, num_guesses));
  return 0;
}
